File: 605. Can Place Flowers/main.cpp
#include <vector>

using namespace std;

class Solution {
    bool isEmpty(vector<int> &flowerbed, int pos) {
        return pos < 0 || pos >= flowerbed.size() || flowerbed[pos] == 0;
    }

public:
    bool canPlaceFlowers(vector<int> &flowerbed, int n) {
        int count = 0;

        for (size_t i = 0; i < flowerbed.size(); i++) {
            if (isEmpty(flowerbed, i - 1) && isEmpty(flowerbed, i) && isEmpty(flowerbed, i + 1)) {
                count++;
                flowerbed[i] = 1;
            }
        }

        return count >= n;
    }
};

class Solution {
public:
    bool canPlaceFlowers(vector<int> &flowerbed, int n) {
        int count = 0;
        int size = flowerbed.size();

        for (int i = 0; i < flowerbed.size(); i++) {
            bool left = i - 1 < 0 || flowerbed[i - 1] == 0;
            bool right = i + 1 >= size || flowerbed[i + 1] == 0;

            if (left && right && flowerbed[i] == 0) {
                count++;
                flowerbed[i] = 1;
            }
        }

        return count >= n;
    }
};

File: 605. Can Place Flowers/main.py
from typing import List


class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        def isEmpty(pos: int) -> bool:
            return pos < 0 or pos >= len(flowerbed) or flowerbed[pos] == 0

        count = 0

        for i in range(len(flowerbed)):
            if isEmpty(i - 1) and isEmpty(i) and isEmpty(i + 1):
                count += 1
                flowerbed[i] = 1

        return count >= n


class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        count = 0

        for i in range(len(flowerbed)):
            left = i - 1 < 0 or flowerbed[i - 1] == 0
            right = i + 1 >= len(flowerbed) or flowerbed[i + 1] == 0

            if left and right and flowerbed[i] == 0:
                count += 1
                flowerbed[i] = 1

        return count >= n
